### DISCUSSION TITLE
Make it easier to use contentment radio and checkbutton lists in configuration of data sources

### DISCUSSION THREAD

**Discussion by erlando** [leekelleher/umbraco-contentment#362] [Idea]
Internally Contentment uses its own editors for config. This makes complete sense. Contentment has constants for configuration field aliases and editor view paths but alas they are internal and cannot be used by 3rd party implementations.

I would like to suggest that Constants.Conventions.ConfigurationFieldAliases be made publicly accessible.

It would also be nice if the View member had a static pendant such that you do not have to do things like this 
`View = _ioHelper.ResolveRelativeOrVirtualUrl(new CheckboxListDataListEditor().View)` when setting up a configuration field for a data source. It just seems.... wrong somehow :-)

Lastly, thank you for a very nice package.

**Comment by leekelleher** [comment]
Thank you for raising this @erlando.  In recent years, my general stance with my code is to mark it `internal` (and often `sealed`), then when people ask for it to be opened up, then review from there.  This is mostly to reduce my own technical debt, as once `public`, who knows how developers are using it.

Thanks for the prompt, I'm happy to open up some of the constants.

I'll give it some thought on how to best go about it, as my initial thought is not to change the [`Constants`](https://github.com/leekelleher/umbraco-contentment/blob/4.5.0/src/Umbraco.Community.Contentment/Constants.cs) class from `internal` to `public`... I'm thinking along the lines of introducing a new static class `ContentmentConstants`, mostly to avoid naming conflicts with Umbraco's own [`Constants` container class](https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Core/Constants-PropertyEditors.cs).

**Reply by leekelleher** [reply]
@erlando Following up on this _(a few months later),_ I've got a pull request ready #375 to add [a `ContentmentConstants` class](https://github.com/leekelleher/umbraco-contentment/blob/f1e2cd9daf9bf1499a5ef5123e52bc1e115a00cd/src/Umbraco.Community.Contentment/ContentmentConstants.cs) to expose certain internal constants.

So far it'll have the property-editor aliases, view paths and some of the popular configuration keys/aliases.

Let me know if anything else useful is missing from there.  I'll be happy to add more in future _(if it makes sense)._

The PR will target the next major version, v5 - _mostly because that's where my development efforts are currently focused._

**Reply by leekelleher** [reply]
I've merged in PR #375 for Contentment v5.

**Reply by erlando** [reply]
Awesome. Thank you for your continued work on this excellent package.
